src/frontend/Pretty_printing.ml

```diff
@@ -210,29 +210,32 @@ and pp_transformation ppf = function
   | Correlation -> Fmt.pf ppf ""
   | Covariance -> Fmt.pf ppf ""
 
-and pp_transformed_type ppf (st, trans) =
-  let unsizedtype_fmt, sizes_fmt =
-    match st with
-    | Middle.SInt -> (Fmt.const pp_unsizedtype UInt, Fmt.nop)
-    | SReal -> (Fmt.const pp_unsizedtype UReal, Fmt.nop)
-    | SVector e ->
-        ( Fmt.const pp_unsizedtype UVector
-        , Fmt.const (fun ppf -> Fmt.pf ppf "[%a]" pp_expression) e )
-    | SRowVector e ->
-        ( Fmt.const pp_unsizedtype URowVector
-        , Fmt.const (fun ppf -> Fmt.pf ppf "[%a]" pp_expression) e )
-    | SMatrix (e1, e2) ->
-        ( Fmt.const pp_unsizedtype UMatrix
-        , Fmt.const
-            (fun ppf -> Fmt.pf ppf "[%a, %a]" pp_expression e1 pp_expression)
-            e2 )
-    | SArray _ -> (
-      match unwind_sized_array_type st with st, _ ->
-        (Fmt.const pp_sizedtype st, Fmt.nop) )
+and pp_transformed_type ppf (pst, trans) =
+  let pst =
+    match pst with
+    | Middle.Sized st ->
+        Middle.Sized (Fn.compose fst unwind_sized_array_type st)
+    | Unsized ut -> Middle.Unsized ut
```

Comment:
I think this is wrong. We seem to want to pull out the base type (discarding all array constructors here).

Reply:
I still don't get why this is written like this... seems like you'd want each pretty printer to be modular; e.g. unsizedtypes can always be emitted whole by pp_unsizedtype :P Oh well, I'll fix this locally for now.